pkg/operator/nsfinalizercontroller/finalizer_controller.go

```diff
@@ -68,7 +73,7 @@ func NewFinalizerController(
 }
 
 func (c finalizerController) sync() error {
```

Comment:
so this controller simply removes all finalizers except `FinalizerKubernetes` from a namespace?

Reply:
> so this controller simply removes all finalizers except `FinalizerKubernetes` from a namespace?

It removes only the kube finalizer.  The kube finalizer requires every aggregated apiserver be running. When you're removing an aggregated apiserver's namespace that will come back, you don't want to remove the apiservices or controllers will do bad things, so this allows the namespace to be removed and recreated.